cas-client-core/src/main/java/org/jasig/cas/client/util/ThreadLocalXPathExpression.java

```diff
@@ -0,0 +1,87 @@
+package org.jasig.cas.client.util;
+
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+import javax.xml.namespace.NamespaceContext;
+import javax.xml.namespace.QName;
+import javax.xml.xpath.*;
+
+/**
+ * Thread local XPath expression.
+ *
+ * @author Marvin S. Addison
+ * @since 3.3.4
+ */
+public class ThreadLocalXPathExpression extends ThreadLocal<XPathExpression> implements XPathExpression {
```

Comment:
If I take a closer look, I only see `expression` and `context` as shared variables and they seem to be accessed only for reading, don't they? Shouldn't they be declared `final`as well?
I guess this line: `XPathFactory.newInstance().newXPath()` is the one which takes time to initialize?